<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="manipulator">
  <link name="world">
   </link>
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="0.0021253 0.047935 0.027764"
        rpy="0 0 0" />
      <mass
        value="0.0896" />
      <inertia
        ixx="5.041E-05"
        ixy="-8.5548E-08"
        ixz="3.2278E-08"
        iyy="6.5382E-05"
        iyz="4.1605E-12"
        izz="0.00010836" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_rviz/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.62745 0.62745 0.62745 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_rviz/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="dummy_joint" type="fixed">
     <parent link="world"/>
     <child link="base_link"/>
   </joint>
  <link
    name="link_1">
    <inertial>
      <origin
        xyz="0.00097841 -0.015906 0.017371"
        rpy="0 0 0" />
      <mass
        value="0.024846" />
      <inertia
        ixx="3.5165E-06"
        ixy="-2.305E-07"
        ixz="-7.8362E-10"
        iyy="7.2755E-06"
        iyz="-1.2738E-08"
        izz="6.9877E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_rviz/meshes/link_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_rviz/meshes/link_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_1"
    type="revolute">
    <origin
      xyz="-0.0030196 0.046937 0.0635"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="link_1" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="1.0"
      velocity="1.0" />
  </joint>
  <link
    name="link_2">
    <inertial>
      <origin
        xyz="-0.00039992 -0.0021855 0.052975"
        rpy="0 0 0" />
      <mass
        value="0.01633" />
      <inertia
        ixx="5.4124E-06"
        ixy="-3.0048E-08"
        ixz="-1.8798E-08"
        iyy="5.9019E-06"
        iyz="-1.3839E-08"
        izz="6.375E-07" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_rviz/meshes/link_2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_rviz/meshes/link_2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_2"
    type="revolute">
    <origin
      xyz="0.0010394 -0.0095198 0.034407"
      rpy="0 0 0" />
    <parent
      link="link_1" />
    <child
      link="link_2" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-1.26"
      upper="1.26"
      effort="1.0"
      velocity="1.0" />
  </joint>
  <link
    name="link_3">
    <inertial>
      <origin
        xyz="0.06228 0.002117 -0.019509"
        rpy="0 0 0" />
      <mass
        value="0.019923" />
      <inertia
        ixx="1.655E-06"
        ixy="-2.7692E-07"
        ixz="1.0523E-06"
        iyy="5.1979E-06"
        iyz="9.2887E-08"
        izz="5.4044E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_rviz/meshes/link_3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_rviz/meshes/link_3.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_3"
    type="revolute">
    <origin
      xyz="-0.00063851 0.011696 0.072593"
      rpy="0 0 0" />
    <parent
      link="link_2" />
    <child
      link="link_3" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-1.87"
      upper="0.80"
      effort="1.0"
      velocity="1.0" />
  </joint>
  <link
    name="claw_right">
    <inertial>
      <origin
        xyz="0.0078791 0.0033483 -0.0010726"
        rpy="0 0 0" />
      <mass
        value="0.043225" />
        <!-- value="0.013225" -->
      <inertia
        ixx="7.6987E-07"
        ixy="-3.6907E-07"
        ixz="6.5483E-07"
        iyy="2.672E-06"
        iyz="1.0553E-07"
        izz="3.0884E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 -0.2 0" />
      <geometry>
        <mesh
          filename="package://simulation_rviz/meshes/claw_right.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.77647 0.75686 0.73725 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0.02"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_rviz/meshes/claw_right.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_4"
    type="revolute">
    <origin xyz="0.091729 -0.0080548 -0.011749"
      rpy="0 0.2 0" />
    <parent
      link="link_3" />
    <child
      link="claw_right" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="0"
      upper="0.9"
      effort="1.0"
      velocity="1.0" />
  </joint>
  <link
    name="claw_left">
    <inertial>
      <origin
        xyz="0.017988 0.013506 0.0076557"
        rpy="0 0 0" />
      <mass
        value="0.013642" />
      <inertia
        ixx="8.1401E-07"
        ixy="-4.7952E-08"
        ixz="6.8628E-07"
        iyy="2.8195E-06"
        iyz="2.246E-08"
        izz="3.2661E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 -0.0131993 0"
        rpy="0 -0.2 0" />
      <geometry>
        <mesh
          filename="package://simulation_rviz/meshes/claw_left.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.77647 0.75686 0.73725 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_rviz/meshes/claw_left.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_5"
    type="revolute">
    <origin
      xyz="0.079643 0.0231993 -0.020398"
      rpy="0 0.2 0" />
    <parent
      link="link_3" />
    <child
      link="claw_left" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="0.9"
      effort="1.0"
      velocity="1.0" />
  </joint>

  <gazebo>
    	<plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
      		<robotNamespace>/manipulator</robotNamespace>
    	</plugin>
  	</gazebo>
	<transmission name="joint_1_trans">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="joint_1">
			<hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
		</joint>
		<actuator name="joint_1_motor">
			<mechanicalReduction>2</mechanicalReduction>
			<hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
		</actuator>
	</transmission>
	<transmission name="joint_2_trans">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="joint_2">
			<hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
		</joint>
		<actuator name="joint_2_motor">
			<mechanicalReduction>2</mechanicalReduction>
			<hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
		</actuator>
	</transmission>
	<transmission name="joint_3_trans">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="joint_3">
			<hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
		</joint>
		<actuator name="joint_3_motor">
			<mechanicalReduction>2</mechanicalReduction>
			<hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
		</actuator>
	</transmission>
	<transmission name="joint_4_trans">
			<type>transmission_interface/SimpleTransmission</type>
			<joint name="joint_4">
				<hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
			</joint>
			<actuator name="joint_4_motor">
				<mechanicalReduction>2</mechanicalReduction>
				<hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
			</actuator>
	</transmission> 
  <transmission name="joint_5_trans">
			<type>transmission_interface/SimpleTransmission</type>
			<joint name="joint_5">
				<hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
			</joint>
			<actuator name="joint_5_motor">
				<mechanicalReduction>2</mechanicalReduction>
				<hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
			</actuator>
	</transmission> 
  
</robot>